geomstats/learning/mdm.py

```diff
@@ -0,0 +1,114 @@
+"""The MDM classifier on manifolds."""
+
+import geomstats.backend as gs
+from geomstats.learning.frechet_mean import FrechetMean
+
+
+class RiemannianMinimumDistanceToMeanClassifier:
+    """
+    Classifier implementing the MDM scheme on manifolds.
+
+    Attributes
+    ----------
+    riemannian_metric : string or callable.
+        The distance metric to use.
+    n_clusters: int, number of clusters.
+    mean_estimates_: array-like, shape=[n_classes, n_features] if
+    point_type='vector'
+                   shape=[n_classes, n_features, n_features]
+                    if point_type='matrix'
+       Frechet means of each class.
+    """
+
+    def __init__(
+            self,
+            riemannian_metric,
+            n_clusters,
+            point_type='matrix'):
+        self.riemannian_metric = riemannian_metric
+        self.n_clusters = n_clusters
+        self.point_type = point_type
+        self.mean_estimates_ = None
+
+    def fit(self, X, y):
+        """
+        Compute Frechet mean of each class.
+
+        Stores the mean in the class attributes mean_estimates_.
+
+        Parameters
+        ----------
+        X : array-like, shape=[n_samples, n_features]
+                              if point_type='vector'
+                              shape=[n_samples, n_features, n_features]
+                              if point_type='matrix'
+                  Training data, where n_samples is the number of samples
+                  and n_features is the number of features.
+        y : array-like, shape=[n_samples, n_classes]
+                  Training labels, where n_classes is the number of classes.
+        """
+        mean_estimator = FrechetMean(
+            metric=self.riemannian_metric,
+            point_type=self.point_type)
+        frechet_means = []
+        for c in range(self.n_clusters):
+            data_class = self.split_data_in_classes(X, y, c)
+            # frechet_means.append(mean_estimator.fit(data_class).estimate_[0])
+            frechet_means.append(mean_estimator.fit(data_class).estimate_)
+        self.mean_estimates_ = gs.array(frechet_means)
+
+    def predict(self, X):
+        """
+        Compute closest neighbor according to riemannian_metric.
+
+        Parameters
+        ----------
+        X : array-like, shape=[n_samples, n_features]
+                              if point_type='vector'
+                              shape=[n_samples, n_features, n_features]
+                              if point_type='matrix'
```

Comment:
This may not be the case in the rest of the code base, but I don't think we should use `n_features`, but `dim` instead, as this carries more the geometric attribute of the manifold: features are independent parameters, shaping a vector space.
For the case `point_type='matrix'`, we usually use `[n, n]` with no real meaning